### PR TITLE
Hobbes to Selenium Migration

### PR DESCRIPTION
**Pull request by absarasw**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Minor: New Feature?      |
| Major: Breaking Change?  |
| Tests Added + Pass?      | Yes
| Documentation Provided   | Yes (code comments and or markdown)
| Any Dependency Changes?  |
| License                  | Apache License, Version 2.0

### CODE REVIEW

File: testing/it/e2e-selenium/src/test/java/com/adobe/cq/wcm/core/components/it/seljup/components/Commons/ChildrenEditor.java [hunk 1]
```diff
@@ -67,6 +67,7 @@ public void removeFirstItem() {
      * @throws InterruptedException
      */
     public void moveItems(int dragElement,int targetElement) throws InterruptedException {
+        Commons.webDriverWait(1000);
```
Comment (vladbailescu) [inline testing/it/e2e-selenium/src/test/java/com/adobe/cq/wcm/core/components/it/seljup/components/Commons/ChildrenEditor.java:70]: I'd rather we define a constant for this.

Comment (absarasw) [reply testing/it/e2e-selenium/src/test/java/com/adobe/cq/wcm/core/components/it/seljup/components/Commons/ChildrenEditor.java:70]: Yes, I'll update that.

**Comment by msagolj** [comment]
Some general questions/remarks:

1. There are inconsistencies in this PR and in the branch regarding package names:
- java package names should always be lowercase, which is always the case except for the package names below `/com/adobe/cq/wcm/core/components/it/seljup/components`, there all the sub packages are uppercase.
- same for package names `v1` and `v2`. They are all lowercase but in this PR there are some which are uppercase e.g.:
`package com.adobe.cq.wcm.core.components.it.seljup.tests.FormText.V2;`

2. A lot of times the version V1 or V2 is part of the class name for component and test classes even tough its already clear from the package which version it is e.g. 
`com.adobe.cq.wcm.core.components.it.seljup.components.FormButton.v1.FormbuttonV1`
`com.adobe.cq.wcm.core.components.it.seljup.tests.accordion.v1.AccordionV1IT`
is there a specific reason for it  ? if so it should be consistent, Accordion, Button and Carousel component classes don't have  version in the name. I would rather go for removing them from the class name if possible.

3. Most of he different versions of a component class (e.g. `FormButtonV1`, `FormButtonV2`) extend from a  common base class (e.g. `FormButton`).Maybe rename this one to `BaseFormButton` or if possible even make it abstract (and call it `AbstractFormButton`) to make clear its not to be used directly.

4. Inconsistent Naming:  the dialog class is sometimes called `...ConfigureDialog` (e.g. `AccordionConfigureDialog`, sometimes  `...ConfigDialog` (e.g. `BreadcrumbConfigDialog`). Maybe rename it to `..EditDialog` as this is how they are usually referred to.

**Comment by msagolj** [comment]
The Dialog class:
`com.adobe.cq.wcm.core.components.it.seljup.components.button.v1.ButtonConfigureDialog`
should probably be one level higher in :  
`com.adobe.cq.wcm.core.components.it.seljup.components.button.ButtonConfigureDialog`
like the other dialog classes.

**Comment by msagolj** [comment]
We should group the tests so they can run in parallel on Jenkins as the ~50 tests we have now already take up to 30 minutes to run (https://jenkins-sprouts.ci.corp.adobe.com/view/Ref-Squad/job/github-aem-core-wcm-components-hobbes-to-selenium)

To put a test in a group you add a `@Tag('groupname')` on top of a test class e.g. https://git.corp.adobe.com/CQ/ui-wcm-editor/blob/master/it/cq-ui-wcm-editor-it-e2e-seljup/src/test/java/com/adobe/cq/wcm/core/authoring/it/seljup/pageeditor/SanityIT.java#L38

Maybe start with 2 groups, e.g. tag all Form Component related test classes with `@Tag('group1')`, and the rest with `@Tag('group2')`.

**Comment by msagolj** [comment]
The breadcrumb tests are the only tests where the actually test method implementations are extracted into a separate class:

`com.adobe.cq.wcm.core.components.it.seljup.tests.breadcrumb.BreadcrumbTests`

while the others have the implementation either in the v1 or v2 version of the test class.  What's the reasoning ?

**Comment by msagolj** [comment]
The README.md at :

`aem-core-wcm-components\testing\it\e2e-selenium\README.md`

should be updated. I guess it was intially cloned from a different project as it mentions `WCM Launches UI testing` and not all of the content applies to this module.

**Comment by absarasw** [comment]
> Some general questions/remarks:
> 
> 1. There are inconsistencies in this PR and in the branch regarding package names:
> 
> * java package names should always be lowercase, which is always the case except for the package names below `/com/adobe/cq/wcm/core/components/it/seljup/components`, there all the sub packages are uppercase.
> * same for package names `v1` and `v2`. They are all lowercase but in this PR there are some which are uppercase e.g.:
>   `package com.adobe.cq.wcm.core.components.it.seljup.tests.FormText.V2;`
> 

Changing it to lowercase package name. 
> 1. A lot of times the version V1 or V2 is part of the class name for component and test classes even tough its already clear from the package which version it is e.g.
>    `com.adobe.cq.wcm.core.components.it.seljup.components.FormButton.v1.FormbuttonV1`
>    `com.adobe.cq.wcm.core.components.it.seljup.tests.accordion.v1.AccordionV1IT`
>    is there a specific reason for it  ? if so it should be consistent, Accordion, Button and Carousel component classes don't have  version in the name. I would rather go for removing them from the class name if possible.

For component class we can remove the version but for test classes we need to differentiate since for running the test we need to mention the class name and we cannot have 2 class with same name.

> 2. Most of the different versions of a component class (e.g. `FormButtonV1`, `FormButtonV2`) extend from a  common base class (e.g. `FormButton`).Maybe rename this one to `BaseFormButton` or if possible even make it abstract (and call it `AbstractFormButton`) to make clear its not to be used directly.

I think naming it Base<component-name> would be fine as it has most of the methods defined in it. The version classes only contain the selectors.
> 3. Inconsistent Naming:  the dialog class is sometimes called `...ConfigureDialog` (e.g. `AccordionConfigureDialog`, sometimes  `...ConfigDialog` (e.g. `BreadcrumbConfigDialog`). Maybe rename it to `..EditDialog` as this is how they are usually referred to.
Changing the name as `*ConfigDialog` for all components.

**Comment by absarasw** [comment]
> The breadcrumb tests are the only tests where the actually test method implementations are extracted into a separate class:
> 
> `com.adobe.cq.wcm.core.components.it.seljup.tests.breadcrumb.BreadcrumbTests`
> 
> while the others have the implementation either in the v1 or v2 version of the test class. What's the reasoning ?

I did that since breadcrumb V2 has some extra tests, so I thought defining the tests separately would help. But now I think that wasn't required. The tests can be defined in version IT classes. 
@msagolj  Should I move the tests to version classes?

File: testing/it/e2e-selenium/src/test/java/com/adobe/cq/wcm/core/components/it/seljup/components/embed/UrlProcessors.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    public class OEmbed {
+        private String name;
+        private String selector;
+        private String [] urls;
+
+        public String getName() {
+            return name;
+        }
+
+        public boolean isUrlProcessorExits() {
```
Comment (jckautzmann) [inline testing/it/e2e-selenium/src/test/java/com/adobe/cq/wcm/core/components/it/seljup/components/embed/UrlProcessors.java:33]: Is this method checking for existence of URL processor? If so, I'd rename it `urlProcessorExists`.

File: testing/it/e2e-selenium/src/test/java/com/adobe/cq/wcm/core/components/it/seljup/components/embed/UrlProcessors.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            Helpers.waitForElementAnimationFinished(selector);
+            return $(selector).isDisplayed();
+        }
+
+        public String []  getUrls() {
```
Comment (jckautzmann) [inline testing/it/e2e-selenium/src/test/java/com/adobe/cq/wcm/core/components/it/seljup/components/embed/UrlProcessors.java:38]: I'd trim unnecessary whitespaces: `public String[] getUrls() {`

**Pull request merged**